Jenkinsfile

```
@@ -16,215 +16,211 @@ pipeline {
-            changeset "src/main/**"
-            triggeredBy cause: 'UserIdCause'
-          }
-          expression {
-            return env.GIT_BRANCH == 'origin/1.1.0';
+
+        stage('Start acl-apd-Server for Integration Testing'){
+          steps{
+            script{
+              sh 'sudo update-alternatives --set java /usr/lib/jvm/java-21-openjdk-amd64/bin/java'
+              sh 'scp src/test/resources/DX-ACL-APD-APIs.postman_collection.json jenkins@jenkins-master:/var/lib/jenkins/iudx/acl-apd/Newman/'
+              sh 'mvn flyway:migrate -Dflyway.configFiles=/home/ubuntu/configs/acl-apd-flyway.conf'
+              sh 'docker compose -f docker-compose.test.yml up -d integTest'
+              sh 'sleep 45'
+            }
           }
-        }
-      }
-      stages {
-        stage('Push Images') {
-          steps {
-            script {
-              docker.withRegistry( registryUri, registryCredential ) {
-                devImage.push("1.2.0-alpha-${env.GIT_HASH}")
-                deplImage.push("1.2.0-alpha-${env.GIT_HASH}")
+          post{
+            failure{
+              script{
+                sh 'mvn flyway:clean -Dflyway.configFiles=/home/ubuntu/configs/acl-apd-flyway.conf'
+                sh 'docker compose -f docker-compose.test.yml down --remove-orphans'
               }
+              cleanWs deleteDirs: true, disableDeferredWipeout: true
             }
           }
         }
-        stage('Docker Swarm deployment') {
-          steps {
-            script {
-              sh "ssh azureuser@docker-swarm 'docker service update acl-apd_acl-apd --image ghcr.io/datakaveri/acl-apd-depl:1.2.0-alpha-${env.GIT_HASH}'"
-              sh 'sleep 40'
-              sh '''#!/bin/bash
-                response_code=$(curl -s -o /dev/null -w "%{http_code}" --connect-timeout 5 --retry 5 --retry-connrefused -XGET https://acl-apd.iudx.io/apis)
-                echo $response_code
-                if [[ "$response_code" -ne "200" ]]
-                then
-                  echo "Health check failed"
-                  exit 1
-                else
-                  echo "Health check complete; Server is up."
-                  exit 0
-                fi
-              '''
+
+        stage('Integration Tests and OWASP ZAP pen test'){
+          steps{
+            node('built-in') {
+              script{
+                startZap ([host: 'localhost', port: 8090, zapHome: '/var/lib/jenkins/tools/com.cloudbees.jenkins.plugins.customtools.CustomTool/OWASP_ZAP/ZAP_2.11.0'])
+                sh 'curl http://127.0.0.1:8090/JSON/pscan/action/disableScanners/?ids=10096'
+                sh 'HTTP_PROXY=\'127.0.0.1:8090\' newman run /var/lib/jenkins/iudx/acl-apd/Newman/DX-ACL-APD-APIs.postman_collection.json -e /home/ubuntu/configs/5.6.0/acl-apd-postman-env.json --insecure -r htmlextra --reporter-htmlextra-export /var/lib/jenkins/iudx/acl-apd/Newman/report/report.html --reporter-htmlextra-skipSensitiveData'
```

The master pipeline should not be using 5.6.0 env
```suggestion
                sh 'HTTP_PROXY=\'127.0.0.1:8090\' newman run /var/lib/jenkins/iudx/acl-apd/Newman/DX-ACL-APD-APIs.postman_collection.json -e /home/ubuntu/configs/acl-apd-postman-env.json --insecure -r htmlextra --reporter-htmlextra-export /var/lib/jenkins/iudx/acl-apd/Newman/report/report.html --reporter-htmlextra-skipSensitiveData'
```